src/plugins_types.c

```diff
@@ -685,7 +685,8 @@ lyplg_type_check_hints(uint32_t hints, const char *value, size_t value_len, LY_D
     case LY_TYPE_INT32:
         LY_CHECK_ARG_RET(NULL, base, LY_EINVAL);
 
-        if (!(hints & (LYD_VALHINT_DECNUM | LYD_VALHINT_OCTNUM | LYD_VALHINT_HEXNUM))) {
+        if (!(hints & (LYD_VALHINT_DECNUM | LYD_VALHINT_OCTNUM | LYD_VALHINT_HEXNUM)) &&
+            !(hints & LYD_VALHINT_STRING_DATATYPES)) {
```

Comment:
Coding style issues, we use double indent when continuing the previous line.

Reply:
fixed

Reply:
Ever think about adding a .clang-format to help people with code formatting?  Its gotten pretty good these days, we use it in c-ares :)

Reply:
We tried that but strongly prefer `uncrustify`, we could not configure `clang` for our coding style. But you can use it to format the auotmatically, even directly supported by our projects, just run `make format`.

Reply:
ah, good to know, didn't see that

and yeah, clang-format works great on most code styles, but if you have a deviation you have to submit a patch to clang and add a toggle for that.  Actually, I recently upstreamed one for c-ares.